### ISSUE TITLE
Mass-curate OLS redirects, e.g. nmrCV

### ISSUE THREAD

**Issue by sneumann** [identifiers-org/identifiers-org.github.io#250]
Hi,

I found an issue on https://registry.identifiers.org/registry/nmr
and suspect there could be something systematic, and if so, 
this could be mass-curated for other affected identifiers.org prefixes.

For example, https://identifiers.org/NMR:1000003
redirects to 
https://www.ebi.ac.uk/ols4/ontologies/nmrcv/terms?short_form=NMR:1000003
which in turn gives a "not found" redirecting to 
https://www.ebi.ac.uk/ols4/api/v2/ontologies/nmrcv/classes?shortForm=NMR:1000003&includeObsoleteEntities=true

I can't tell where the underlying issue is (the identifiers.org entry or the OLS4 side ?!),
the correct URL after all redirection would be 
https://www.ebi.ac.uk/ols4/ontologies/nmrcv/classes/http%253A%252F%252FnmrML.org%252FnmrCV%2523NMR%253A1000003

and maybe @henrietteharmse could chime in. 

Thanks,
Yours,
Steffen

**Comment by renatocjn** [comment]
Hi Steffen,
When OLS moved to OLS4, there were issues with some ontologies. I opened [this ticket](https://github.com/EBISPOT/ols4/issues/388) at the time but never got around to fixing the remaining ontologies because I was waiting for any further evolutions and other tasks came up. Thank you for reminding me. I see some ontologies still result in not found, any help from the OLS team would be appreciated.

**Comment by renatocjn** [comment]
Current list of namespaces with OLS resources: bao, biomodels.kisao, biomodels.teddy, bto, cco, chebi, cheminf, cl, clo, doid, eco, edam, efo, envo, eo, fma, foodon, go, hp, ido, ma, mamo, maxo, mcro, mi, mod, mp, ms, ncit, nmr, obcs, obi, omit, opl, orphanet.ordo, pato, po, pr, probonto, pw, ro, sbo, sio, so, stato, taxonomy, uberon, uo, vario

**Comment by henrietteharmse** [comment]
@sneumann The issue is with the short_form specified. Short_forms has this format: `NMR_1000003`. That is it should use an `_` rather than a `:`. 

Hence, this works: [https://www.ebi.ac.uk/ols4/ontologies/nmrcv/classes?short_form=NMR_1000003](https://www.ebi.ac.uk/ols4/ontologies/nmrcv/classes?short_form=NMR_1000003).

If you prefer to use curies, you could use: [https://www.ebi.ac.uk/ols4/ontologies/nmrcv/classes?curie=NMR:1000003](https://www.ebi.ac.uk/ols4/ontologies/nmrcv/classes?curie=NMR:1000003).

**Comment by renatocjn** [comment]
Thank you for the information @henrietteharmse !

We, unfortunately, can't use underscores since identifiers.org only allows for the colon or the slash as separating characters.

Is the curie query parameter available for all ontologies?

**Comment by henrietteharmse** [comment]
Yes, the curie parameter is available for all ontologies!

**Comment by henrietteharmse** [comment]
Yes, there is indeed several terms in the EDAM ontology that matches on Miriam. See [EDAM ontology source](https://raw.githubusercontent.com/edamontology/edamontology/master/releases/EDAM.owl).

### Use IRI
The fail-safe way to get this consistently from OLS for EDAM is to use an IRI. I.e. in this case the IRI you are interested in is 
http://edamontology.org/data_1164.

To access the OLS UI, use the following where the IRI is double URL encoded:
https://www.ebi.ac.uk/ols4/ontologies/edam/classes/{double urlencoded IRI}

==>
https://www.ebi.ac.uk/ols4/ontologies/edam/classes/http%253A%252F%252Fedamontology.org%252Fdata_1164

Or, if you are using the API use:
https://www.ebi.ac.uk/ols4/api/ontologies/edam/terms/{double urlencode IRI}

https://www.ebi.ac.uk/ols4/api/ontologies/edam/terms/http%253A%252F%252Fedamontology.org%252Fdata_1164 

### Use Curie
If you want to use curies, you can use the following:

For UI: https://www.ebi.ac.uk/ols4/ontologies/edam/classes?curie=EDAM:1164
For API: https://www.ebi.ac.uk/ols4/api/ontologies/edam/terms?curie=EDAM:1164

However, with curies there are some ambiguity.

**Comment by renatocjn** [comment]
Awesome, thank you for all the help @henrietteharmse 

@sneumann may I close the issue?

**Comment by sneumann** [comment]
https://identifiers.org/NMR:1000003 now works as expected, thanks for fixing ! Yours, Steffen
